Content.Client/UserInterface/ContextMenu/ContextMenuView.cs

```diff
@@ -0,0 +1,265 @@
+﻿#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Robust.Client.UserInterface;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Maths;
+
+namespace Content.Client.UserInterface.ContextMenu
+{
+    public interface IContextMenuView : IDisposable
+    {
+        Dictionary<IEntity, ContextMenuElement> Elements { get; set; }
+        Stack<ContextMenuPopup> Menus { get; }
+        event EventHandler<(GUIBoundKeyEventArgs, SingleContextElement)>? OnKeyBindDownSingle;
+        event EventHandler<SingleContextElement>? OnMouseEnteredSingle;
+        event EventHandler<SingleContextElement>? OnMouseExitedSingle;
+        event EventHandler<SingleContextElement>? OnMouseHoveringSingle;
+
+        event EventHandler<(GUIBoundKeyEventArgs, StackContextElement)>? OnKeyBindDownStack;
+        event EventHandler<StackContextElement>? OnMouseEnteredStack;
+
+        event EventHandler<ContextMenuElement>? OnExitedTree;
+
+        event EventHandler? OnCloseRootMenu;
+        event EventHandler<int>? OnCloseChildMenu;
```

Comment:
paging @RemieRichards since i'm not deep in the whole event talk, is this fine or should they use something else?

Reply:
I think it's fine.
We've got eventhandlers in a few places.